frontend/lib/dev-server.js

```diff
@@ -0,0 +1,58 @@
+const fs = require('fs');
+const url = require('url');
+const path = require('path');
+const https = require('https');
+const express = require('express');
+const morgan = require('morgan');
+
+// TODO: CSP copied from bin/deploy.sh - find a way to extract automatically?
+const CSP = `default-src 'self'; connect-src 'self' https://sentry.prod.mozaws.net https://www.google-analytics.com https://ssl.google-analytics.com https://basket.mozilla.org https://analysis-output.telemetry.mozilla.org; font-src 'self' code.cdn.mozilla.net; form-action 'none'; frame-ancestors 'self' https://pontoon.mozilla.org; img-src 'self' https://pontoon.mozilla.org https://ssl.google-analytics.com https://www.google-analytics.com; object-src 'none'; script-src 'self' https://pontoon.mozilla.org https://ssl.google-analytics.com; style-src 'self' https://pontoon.mozilla.org code.cdn.mozilla.net; report-uri /__cspreport__; frame-src https://www.youtube.com;`;
+
+const DevServerMiddleware = (req, res, next) => {
+  const parsed = url.parse(req.url);
+  const { pathname } = parsed;
+
+  // If no dot and trailing slash, try rewriting /path to /path/index.html
+  if (
+    pathname.indexOf('.') === -1 &&
+    pathname.substring(pathname.length - 1) !== '/'
+  ) {
+    parsed.pathname = pathname + '/index.html';
+    req.url = url.format(parsed);
+  }
+
+  if (pathname.indexOf('.json') !== -1) {
```

Comment:
Instead of `.indexOf('.json') !== -1 {}`, should we use ES6'esque [`String#endsWith()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith), or Node [`path.extname()`](https://nodejs.org/dist/latest-v8.x/docs/api/path.html#path_path_extname_path)?